### PR TITLE
Create InputGroup component

### PR DESCRIPTION
**Pull request by HDinger**
This creates a new component `InputGroup` which consists of: 

* A text field
* An trailing action shown next to it. This can either be an icon button or a clipboardCopyButton

Bildschirmfoto 2023-11-24 um 11 32 48

https://community.openproject.org/projects/openproject/work_packages/50741/activity

### CODE REVIEW

File: app/components/primer/open_project/input_group.rb [hunk 1]
```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+module Primer
+  module OpenProject
+    # Add a general description of component here
+    # Add additional usage considerations or best practices that may aid the user to use the component correctly.
+    # @accessibility Add any accessibility considerations
```
Comment (akabiru) [inline app/components/primer/open_project/input_group.rb:7]:
```suggestion
    # `InputGroup` is composed of a text field input with a trailing action
```

File: app/components/primer/open_project/input_group.rb [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    class InputGroup < Primer::Component
+      status :open_project
+
+      # A component that will render to the right of the label.
+      #
+      # To render a clipboardCopyButton, call the `with_trailing_action_clipboard_copy_button` method, which accepts the arguments accepted by <%= link_to_component(Primer::Beta::ClipboardCopyButton) %>.
+      #
+      # To render an iconButton, call the `with_trailing_action_icon` method, which accepts the arguments accepted by <%= link_to_component(Primer::Beta::Octicon) %>.
+      renders_one :trailing_action, types: {
+        clipboard_copy_button: lambda { |**system_arguments, &block|
```
Comment (akabiru) [inline app/components/primer/open_project/input_group.rb:17]: Nice application of polymorphic slots ✨

**Review by akabiru — COMMENTED** [review]
Top work @HDinger , tests look great- only one remark

- [ ] Should we add `input_width` support in this PR? I couldn't find a reference for this in this PR

**Comment by HDinger** [comment]
> Top work @HDinger , tests look great- only one remark
> 
>     * [ ]  Should we add `input_width` support in this PR? I couldn't find a reference for this in this PR

Good point, @akabiru 👍 I added the support for the `input_width`, and added it to the playground.

**Review by akabiru — APPROVED** [review]
🥇 Thank you! 👏🏾

**Comment by HDinger** [comment]
Merging as the lint issue issue is coming from upstream

**Pull request merged**